src/Symfony/Bridge/Twig/Extension/CodeExtension.php

```diff
@@ -151,7 +152,7 @@ public function fileExcerpt($file, $line, $srcContext = 3)
             }
 
             for ($i = max($line - $srcContext, 1), $max = min($line + $srcContext, count($content)); $i <= $max; ++$i) {
-                $lines[] = '<li'.($i == $line ? ' class="selected"' : '').'><div class="anchor" id="line'.$i.'"></div><code>'.self::fixCodeMarkup($content[$i - 1]).'</code></li>';
+                $lines[] = '<li'.($i == $line ? ' class="selected"' : '').'><a class="anchor" name="line'.$i.'"></a><code>'.self::fixCodeMarkup($content[$i - 1]).'</code></li>';
```

Comment:
I see. Maybe generate a hash of `file+line`.. or something, and only add a unique prefix for the 2nd and upcoming instances (ie. the first instance is predictable).